File: src/main/java/org/motivepick/config/ServerConfig.java
package org.motivepick.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

// TODO: check if works with Spring
@Configuration
class ServerConfig {
    @Value("${enforce.https.for.oauth}")
    boolean enforceHttpsForOauth;

    @Value("${authentication.success.url.web}")
    String authenticationSuccessUrlWeb;

    @Value("${authentication.success.url.mobile}")
    String authenticationSuccessUrlMobile;

    @Value("${logout.success.url}")
    String logoutSuccessUrl;

    @Value("${cookie.domain}")
    String cookieDomain;
}

File: src/main/kotlin/org/motivepick/config/ServerConfig.kt
package org.motivepick.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Configuration

@Configuration
class ServerConfig(

    @Value("\${enforce.https.for.oauth}")
    val enforceHttpsForOauth: Boolean,

    @Value("\${authentication.success.url.web}")
    val authenticationSuccessUrlWeb: String,

    @Value("\${authentication.success.url.mobile}")
    val authenticationSuccessUrlMobile: String,

    @Value("\${logout.success.url}")
    val logoutSuccessUrl: String,

    @Value("\${cookie.domain}")
    val cookieDomain: String,

    @Value("\${cookie.secure}")
    val cookieSecure: Boolean,

    @Value("\${cors.allowedOriginPattern}")
    val corsAllowedOriginPattern: String
)
